Sources/Plasma/Apps/plClient/CMakeLists.txt

```diff
@@ -112,12 +112,16 @@ elseif(APPLE)
         Mac-Cocoa/PLSServerStatus.h
     )
     list(APPEND plClient_RESOURCES
+        "$<$<VERSION_GREATER_EQUAL:${XCODE_VERSION},26.0>:Mac-Cocoa/AppIcon.icon>"
         Mac-Cocoa/banner.png
         Mac-Cocoa/banner@2x.png
         Mac-Cocoa/MainMenu.xib
         Mac-Cocoa/PLSLoginWindowController.xib
         Mac-Cocoa/PLSPatcherWindowController.xib
     )
+    set_source_files_properties("Mac-Cocoa/AppIcon.icon" PROPERTIES
+        XCODE_EXPLICIT_FILE_TYPE "folder.iconcomposer.icon"
+    )
```

Comment:
I've added some new CMake that should:

- Make this icon only be included on Xcode 26 (so it's safe to deploy to our existing agents.)
- Set the file type so it gets compiled to the asset catalog correctly by the toolchain.

I think I can call set_source_files_properties even if the file is not included - but my CMake skills are not the greatest.

Reply:
> I'm somewhat unconvinced that just adding it as a resource won't do the right thing.

The issue is that CMake doesn't set the right content type when it generates the project. We have to do that explicitly until CMake is updated (and we get the updated CMake.) I took apart the pbproj to debug the issue.

Reply:
This looks wrong. The source files haven't been added to the target yet, so I feel like this may not work. Worse, `Mac-Cocoa/AppIcon.icon` is a generator expression in the list, but this is unconditional.